### ISSUE TITLE
returning commands strings

### ISSUE THREAD

**Issue by Loumardes** [Red-Teapot/mc-commandblock-1.13-update#5]
Hello
I am looking for a way to handle MCcommands

I am designing a datapack generator in Python (documentation is not the best)

https://github.com/Loumardes/BlocCode

currently MCcommands are inserted as strings arguments in the scriptings functions
if your code allow to intuitively insert commands and return commands strings, i would like to use it

**Comment by Red-Teapot** [comment]
Actually, no, command generation is done via string concatenation, for example:

```python
if len(order) == 1:
    return 'kill @s'
else:
    return 'kill ' + selector.upgrade(props['selector'])
```

Main goal of my project was to parse pre-1.13 commands and generate correct 1.13+ commands that would do the same thing. So, I have a parser but I haven't even tried to implement proper classes for command generation.

**Comment by Loumardes** [comment]
allright
have a look to :
https://github.com/BranislavBajuzik/1.12-to-1.13

also in my project I avoided Object Oriented Programming to try help users to check and inprove source code

**Comment by Red-Teapot** [comment]
I don't think that OOP is bad actually but it's your right to decide whether to use it or not.

Seems like the only thing that might be useful for your BlocCode is the `commands.pre_1_13.nbtstr.types` module as it has an implementation of NBT strings (not actual NBT though) used in commands.
